yaAGC/agc_engine.c

```diff
@@ -1740,64 +1541,1066 @@ SimulateDV(agc_t *State, uint16_t divisor)
+// by F05A, we'll generate an alarm. It also generates HANDRUPTs if any
+// of the input traps are enabled and triggered.
+// Timing pulses GTSET, GTRST, and GTONE are potentially generated,
+// depending on the state of scaler stages 6-9. These pulses are used
+// by the radar, outlink, and altimeter interfaces.
+static void
+TimingSignalF05B(agc_t * State)
+{
+    unsigned GtsetType;
+    // If the input voltage is below the voltage fail circuit threshold,
+    // set a bit for F05A to check.
+    if (State->InputVoltagemV < VFAIL_THRESHOLD)
+        State->InputVoltageLow = 1;
+
+    if (State->Standby)
+        return;
+
+    // Set the PIPA fail bit if any expected PIPA pulses were not received
+    if (State->PipaMissX || State->PipaMissY || State->PipaMissZ)
+      State->InputChannel[033] &= ~010000;
+
+    // If any of the input traps are pending, generate a HANDRUPT and
+    // disable the tripped trap.
+    if (State->Trap31APending)
+    {
+        State->Trap31A = 0;
+        State->Trap31APending = 0;
+        State->InterruptRequests[RUPT_HANDRUPT] = 1;
+    }
+
+    if (State->Trap31BPending)
+    {
+        State->Trap31B = 0;
+        State->Trap31BPending = 0;
+        State->InterruptRequests[RUPT_HANDRUPT] = 1;
+    }
+
+    if (State->Trap32Pending)
+    {
+        State->Trap32 = 0;
+        State->Trap32Pending = 0;
+        State->InterruptRequests[RUPT_HANDRUPT] = 1;
+    }
+
+    // Handled GTSET, GTRST, and GTON-driven logic
+    GtsetType = State->ScalerValue & SCALER_GTSET;
+    if (GtsetType == SCALER_GTSET)
+    {
+        // If the radar gate counter has reached 9, begin generating sync
+        // pulses and getting count requests back.
+        if (State->RadarGateCounter == 9)
+          State->RadarSync = 1;
+
+        // Outlink and tape meter outputs function identically; GTSET will
+        // begin an outgoing pulse sequence if there is not one already
+        // active, otherwise it will end the ongoing one and clear the
+        // channel enable bit.
+        if (State->OutlinkActive)
+        {
+            State->InputChannel[014] &= ~01;
+            State->OutlinkActive = 0;
+        }
+        else if (State->InputChannel[014] & 01)
+        {
+            State->OutlinkActive = 1;
+            State->OutlinkStarting = 1;
+        }
+
+        if (State->AltActive)
+        {
+            State->InputChannel[014] &= ~04;
+            State->AltActive = 0;
+        }
+        else if (State->InputChannel[014] & 04)
+        {
+            State->AltActive = 1;
+            State->AltStarting = 1;
+        }
+    }
+    else if (GtsetType == SCALER_GTRST)
+    {
+        // If a radar sync was active, stop it, generate a RADARUPT, and
+        // reset all of the related state information
+        if (State->RadarSync)
+        {
+            State->InterruptRequests[RUPT_RADARUPT] = 1;
+            State->InputChannel[013] &= ~010;
```

Comment:
This state update needs to propagate to all connected clients / peripheral simulations (via `ChannelOutput` wrapped by `CpuWriteIO`).

```suggestion
            CpuWriteIO(State, 013, ReadIO(State, 013) & ~010);
```

Reply:
@michaelfranzl Are you by chance making these suggestions in the expectation they'll make their way into the main branch of the yaAGC code?  I ask, because your patches are being made against a code branch that it seems clear (from the creator) is not going to be merged soon, if ever, into main.  In the meantime, your patch suggestions are being viewed as something having to do exclusively with this side-branch of the code, and are essentially being ignored as candidates for the inclusion in the main branch of themselves.

Reply:
Thanks for the hint! This is OK for me. :) I'm making suggestions specific to the changes in this branch independent from it being merged, with the intent of making the code slightly more adapted for the integration in peripheral simulations.

As this does create some clutter in the comment section of this pull request, maybe it is better to simply modify my fork of this branch. My changes could then be contributed later.

For what it's worth, the conflicts in this pull request are located only in the comment preambles of a few files and can be easily resolved. My fork of this branch is already rebased on an up-to-date main branch.

Reply:
Oh, I wasn't making any hints about reducing clutter.  The Virtual AGC Project probably seems to many people to be 100% clutter anyway!  :smile:   I just wanted to make sure you weren't having expectations accidentally doomed to failure.  Which you're not.